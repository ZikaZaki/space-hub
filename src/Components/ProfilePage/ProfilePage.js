import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { toggleMission } from '../../redux/missions/missions';
import { toggleRocket } from '../../redux/rockets/rockets';
import { toggleDragon } from '../../redux/dragons/dragons';
// Import styles
import styles from './ProfilePage.module.css';

const ProfilePage = () => {
  const dispatch = useDispatch();
  const reservedRockets = useSelector((state) => state.rockets).filter((rocket) => rocket.reserved);
  const reservedDragons = useSelector((state) => state.dragons).filter((dragon) => dragon.reserved);
  const joinedMissions = useSelector((state) => state.missions).filter((mission) => mission.joined);

  return (
    <div className={styles['page-container']} data-testid="page-container">
      <ul className={styles.list}>
        <span className={styles['list-title']}>My Missions</span>
        { joinedMissions && joinedMissions.map((mission) => (
          <li key={mission.id} className={styles['list-item']}>
            <span className={styles['list-item-title']}>{mission.name}</span>
            {mission.joined && <button type="button" className={styles['cancel-btn']} onClick={() => dispatch(toggleMission(mission.id))}>Leave Mission</button>}
          </li>
        ))}
        { joinedMissions.length === 0 && <span className={styles['list-placeholder']}>...No missions joined</span> }
      </ul>
      <ul className={styles.list}>
        <span className={styles['list-title']}>My Rockets</span>
        { reservedRockets && reservedRockets.map((rocket) => (
          <li key={rocket.id} className={styles['list-item']}>
            <span className={styles['list-item-title']}>{rocket.name}</span>
            {rocket.reserved && <button type="button" className={styles['cancel-btn']} onClick={() => dispatch(toggleRocket(rocket.id))}>Cancel Reservation</button>}
          </li>
        ))}
        { reservedRockets.length === 0 && <span className={styles['list-placeholder']}>...No rockets reserved</span> }
      </ul>
      <ul className={styles.list}>
        <span className={styles['list-title']}>My Dragons</span>
        { reservedDragons && reservedDragons.map((dragon) => (
          <li key={dragon.id} className={styles['list-item']}>
            <span className={styles['list-item-title']}>{dragon.name}</span>
            {dragon.reserved && <button type="button" className={styles['cancel-btn']} onClick={() => dispatch(toggleDragon(dragon.id))}>Cancel Reservation</button>}
          </li>
        ))}
        { reservedDragons.length === 0 && <span className={styles['list-placeholder']}>...No dragons reserved</span> }
      </ul>
    </div>
  );
};

export default ProfilePage;
