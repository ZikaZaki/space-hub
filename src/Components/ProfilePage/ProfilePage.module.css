.page-container {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
  background-color: #fff;
  flex-wrap: wrap;
  gap: 3rem;
  height: 100vh;
  width: 100%;
  padding: 1rem 4rem;
  margin: 0;
}

.list {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  list-style-type: none;
  background-color: rgb(250, 248, 245);
  width: 30%;
  height: auto;
  padding: 0;
  padding-top: 0.5rem;
  margin: 0;
}

.list-title {
  font-weight: bold;
  height: 100%;
  width: 100%;
  padding: 1rem;
  margin: 0;
}

.list-item {
  font-weight: normal;
  color: #121212;
  border: 2px solid #9e9c9c;
  border-bottom: none;
  width: 100%;
  height: 100%;
  padding: 1rem;
  margin: 0;
}

.list-item:last-child {
  border-bottom: 2px solid #9e9c9c;
}

.list-placeholder {
  font-weight: 600;
  color: #9e9c9c;
  width: 100%;
  padding: 1rem;
  margin: 0;
}