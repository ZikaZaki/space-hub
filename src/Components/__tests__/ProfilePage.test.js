import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import ProfilePage from '../ProfilePage/ProfilePage';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component has changed', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <ProfilePage />
    </Provider>,

  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('the component should render the page container', () => {
  render(
    <Provider store={store}>
      <ProfilePage />
    </Provider>,
  );

  const container = screen.getByTestId('page-container');

  expect(container).toBeInTheDocument();
});
