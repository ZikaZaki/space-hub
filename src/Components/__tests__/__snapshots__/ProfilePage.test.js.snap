// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
<div
  className="page-container"
  data-testid="page-container"
>
  <ul
    className="list"
  >
    <span
      className="list-title"
    >
      My Missions
    </span>
    <span
      className="list-placeholder"
    >
      ...No missions joined
    </span>
  </ul>
  <ul
    className="list"
  >
    <span
      className="list-title"
    >
      My Rockets
    </span>
    <span
      className="list-placeholder"
    >
      ...No rockets reserved
    </span>
  </ul>
  <ul
    className="list"
  >
    <span
      className="list-title"
    >
      My Dragons
    </span>
    <span
      className="list-placeholder"
    >
      ...No dragons reserved
    </span>
  </ul>
</div>
`;
